core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -478,6 +478,17 @@ trait SubmissionComponent {
     def confirmInWorkspace(workspaceId: UUID, submissionId: UUID): ReadAction[Option[Unit]] =
       uniqueResult[Unit](findByWorkspaceAndId(workspaceId, submissionId).map(_ => ()))
 
+    def getSubmissionWorkspace(submissionId: UUID): ReadAction[Option[Workspace]] = {
+      // Use a single query with a join between submission and workspace tables
+      val query = for {
+        submission <- submissionQuery if submission.id === submissionId
+        workspace <- workspaceQuery if workspace.id === submission.workspaceId
+      } yield workspace
+
+      // Map the workspace record to a Workspace model object and handle the None case
+      query.result.headOption.map(_.map(WorkspaceRecord.toWorkspace(_)))
```

Comment:
I think this is the cause of the test behavior changes. `WorkspaceRecord.toWorkspace(_)` always generates an empty attribute map, whereas the previous code called `findByName` which does populate the attribute map.

Reply:
I am having a hard time following all the code, but the question is: if you run a submission using this PR's code, and write an output to workspace attributes, will the submission erase any pre-existing attributes?